flink-connectors/flink-connector-mongodb/src/main/java/org/apache/flink/mongodb/table/util/ContextUtil.java

```diff
@@ -0,0 +1,19 @@
+package org.apache.flink.mongodb.table.util;
+
+import org.apache.flink.table.factories.DynamicTableFactory;
+
+import java.util.Map;
+
+public class ContextUtil {
+    public static void transformContext(DynamicTableFactory factory, DynamicTableFactory.Context context) {
+        Map<String, String> catalogOptions = context.getCatalogTable().getOptions();
+
+        Map<String, String> convertedOptions = FactoryOptionUtil.normalizeOptionCaseAsFactory(factory, catalogOptions);
+
+        catalogOptions.clear();
```

Comment:
Modifying the table's options doesn't sound like a good idea, and I don't think it's good we even allow this. The table should probably return an unmodifieable map instead.